lightningd/channel_control.c

```diff
@@ -748,20 +748,24 @@ static void process_check_funding_broadcast(struct bitcoind *bitcoind,
 				    "please consider `close` or `dev-fail`! "));
 		tal_free(cancel->forgets);
 		cancel->forgets = tal_arr(cancel, struct command *, 0);
-		return;
+		goto cleanup;
 	}
 
 	char *error_reason = "Cancel channel by our RPC "
 			     "command before funding "
 			     "transaction broadcast.";
 	forget_channel(cancel, error_reason);
+
+cleanup:
+	tal_free(cc);
+	return;
 }
 
 struct command_result *cancel_channel_before_broadcast(struct command *cmd,
 						       struct peer *peer)
 {
 	struct channel *cancel_channel;
-	struct channel_to_cancel *cc = tal(cmd, struct channel_to_cancel);
+	struct channel_to_cancel *cc = tal(NULL, struct channel_to_cancel);
```

Comment:
Doesn't this now leak if we return with any of the `command_fail` clauses below? If the goal is to have this survive `cmd` it should likely be stolen onto `NULL` and marked as `notleak()` on line 826 instead.

Reply:
Argh good catch thanks..